keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/Nd4jDoubleTensor.java

```diff
@@ -400,6 +431,17 @@ public DoubleTensor applyWhere(DoubleTensor withMask, double value) {
         return this;
     }
 
+    @Override
+    public DoubleTensor applyInPlace(Function<Double, Double> function) {
+        Double[] tensorValues = ArrayUtils.toObject(tensor.toDoubleVector());
+        List<Double> values = Arrays.asList(tensorValues);
+        List<Double> functionApplied = values.stream().map(function::apply).collect(Collectors.toList());
```

Comment:
I love a stream. But because this might be called many many times is there an ugly old school for loop to do this?

Reply:
`King of streams`. Very fair point, updated to be a classic for loop